mono/sgen/sgen-gc.h

```diff
@@ -127,16 +127,40 @@ extern guint64 stat_objects_copied_major;
 		g_error (__VA_ARGS__);	\
 } } while (0)
 
+#ifndef HOST_WIN32
+# define LOG_TIMESTAMP  \
+	do {	\
+		time_t t;									\
+		struct tm tod;									\
+		time(&t);									\
+		localtime_r(&t, &tod);								\
+		strftime(logTime, sizeof(logTime), "%F %T", &tod);				\
+	} while (0)
+#else
+# define LOG_TIMESTAMP  \
+	do {	\
+		time_t t;									\
+		struct tm *tod;									\
+		time(&t);									\
+		tod = localtime(&t);								\
+		strftime(logTime, sizeof(logTime), "%F %T", tod);				\
```

Comment:
Thanks for checking.

From: Johan Lorensson +ADw-notifications+AEA-github.com+ADw-mailto:notifications+AEA-github.com+AD4APg-
Reply-To: mono/mono +ADw-reply++0004b6facd80edea8399f2d7980673e670fcd38597HdtvdnH9NNNNNNNdd+vH+NH/dmte9HHtN3vX3AEA-reply.github.com+ADw-mailto:reply++0004b6facd80edea8399f2d7980673e670fcd38597HdtvdnH9NNNNNNNdd+vH+NH/dmte9HHtN3vX3AEA-reply.github.com+AD4APg-
Date: Friday, June 3, 2016 at 3:30 AM
To: mono/mono +ADw-mono+AEA-noreply.github.com+ADw-mailto:mono+AEA-noreply.github.com+AD4APg-
Cc: Neale Ferguson +ADw-neale+AEA-sinenomine.net+ADw-mailto:neale+AEA-sinenomine.net+AD4APg-, Mention +ADw-mention+AEA-noreply.github.com+ADw-mailto:mention+AEA-noreply.github.com+AD4APg-
Subject: Re: +AFs-mono/mono+AF0- Enhance log tracing (+ACM-2734)


In mono/sgen/sgen-gc.h+ADw-https://github.com/mono/mono/pull/2734+ACM-discussion+AF8-r65666502+AD4-:

+AD4- +ACM- define LOG+AF8-TIMESTAMP  +AFw-
+AD4-       do +AHs-    +AFw-
+AD4-               time+AF8-t t+ADs-                                                                       +AFw-
+AD4-               struct tm tod+ADs-                                                                  +AFw-
+AD4-               time(+ACY-t)+ADs-                                                                       +AFw-
+AD4-               localtime+AF8-r(+ACY-t, +ACY-tod)+ADs-                                                          +AFw-
+AD4-               strftime(logTime, sizeof(logTime), +ACIAJQ-F +ACU-T+ACI-, +ACY-tod)+ADs-                              +AFw-
+AD4-       +AH0- while (0)
+AD4- +ACM-else
+AD4- +ACM- define LOG+AF8-TIMESTAMP  +AFw-
+AD4-       do +AHs-    +AFw-
+AD4-               time+AF8-t t+ADs-                                                                       +AFw-
+AD4-               struct tm +ACo-tod+ADs-                                                                 +AFw-
+AD4-               time(+ACY-t)+ADs-                                                                       +AFw-
+AD4-               tod +AD0- localtime(+ACY-t)+ADs-                                                            +AFw-
+AD4-               strftime(logTime, sizeof(logTime), +ACIAJQ-F +ACU-T+ACI-, tod)+ADs-                               +AFw-


The formatting strings +ACU-F +ACU-T are not supported on WIndows accoring to MSDN:
https://msdn.microsoft.com/en-us/library/fe06s4ak.aspx?f+AD0-255+ACY-MSPPError+AD0--2147217396

but a quick test on VS 2015 c-runtime gave expected results using both +ACU-F and +ACU-T so I believe this will work as expected on windows as well.

+IBQ-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub+ADw-https://github.com/mono/mono/pull/2734/files/583d1a2580a5fc61d0002920c17c23fa71375c3d+ACM-r65666502+AD4-, or mute the thread+ADw-https://github.com/notifications/unsubscribe/AAS2-ggY+AF8-AcC2K1Q1U18Vlw8oTEAv5Kaks5qH9gPgaJpZM4HsGtY+AD4-.